src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,74 @@
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+interface Props {
+  todo: Todo,
+  onCloseBtn: () => void,
+}
+
+export const TodoModal: FC<Props> = (props) => {
+  const {
+    id,
+    userId,
+    title,
+    completed,
+  } = props.todo;
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoadedUser, setIsloadedUser] = useState(false);
+
+  useEffect(() => {
+    getUser(userId)
+      .then(setUser);
+    setIsloadedUser(true);
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+      {(user === null && isLoadedUser)
+        ? <Loader />
+        : (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                {`Todo #${id}`}
+              </div>
+
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => props.onCloseBtn()}
+              />
+            </header>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+            <div className="modal-card-body">
+              <p className="block" data-cy="modal-title">
+                {title}
+              </p>
+
+              <p className="block" data-cy="modal-user">
+                {completed
+                  ? <strong className="has-text-success">Done</strong>
+                  : <strong className="has-text-danger">Planned</strong>}
+
+                {' by '}
+
+                <a href={`mailto:${user?.email}`}>
+                  {user?.name}
```

Comment:
Why user is unnecessary? What will be rendered when there will no any user data?